Adding initial code for LCD to lcd

honglimmq opened the pull request:
-Set up LCD
-Error testing for LCD
-Sample prints to LCD
-Sample print tests to LCD

honglimmq committed:
Adding initial code for LCD to lcd

natlec commented:
Looks good @honglimmq! I'd recommend changing the `UNO.txt` file to markdown so it's a little easier to read through your findings for future reference. Ideally, we should only be committing production code to the master branch, but if you find it'd be useful to share this test code with other members then feel free to pop any test code in a `Testing` subfolder under the `UNO` folder, I reckon the `BTConnector.ino` file can also be moved into its own `Bluetooth` folder under `Testing` as well 🙂

honglimmq commented:
Thanks @natlec! Great feedback. I shall implement your recommendations in my next pull request.

natlec commented:
> Thanks @natlec! Great feedback. I shall implement your recommendations in my next pull request.

No need, feel free to commit the changes to your `lcd` branch - any changes you make to your `lcd` branch will be applied to this PR 😄

TrideepLD reviewed (commented):
Add slightly more detailed comments to things in the ino file on how the components work.

honglimmq committed:
Added scrollLeft() and scrollRight(). Converted all cast of const char* to String. Fixed a few minor bugs. Rearranged and organised sections for better readability.

honglimmq committed:
-Added Button class.
-Refactored code.
-Removed some unnecessary codes.
-Initalised 4 main buttons for use

honglimmq committed:
- Fixed minor bug in displayVelocity()

natlec reviewed (approved):
Looks good to merge 👍
